src/components/CityListScreen.js

```diff
@@ -1,25 +1,74 @@
-import React, { Component } from 'react';
-import { FlatList, View, StyleSheet } from 'react-native';
+import { FlatList, View, StyleSheet, InteractionManager } from 'react-native';
 import { autobind } from 'core-decorators';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
-import * as cityService from '../services/city';
+import { connect } from 'react-redux';
+import _ from 'lodash';
+import { bindActionCreators } from 'redux';
 import CityListItem from './CityListItem';
+import { fetchCurrentWeatherForCity, setCurrentCity, getCities } from '../redux';
 
 class CityListScreen extends Component {
   static propTypes = {
     navigation: PropTypes.object,
+    cities: PropTypes.array,
+    fetchCityWeather: PropTypes.func,
+    onCityPress: PropTypes.func,
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      refreshing: false,
+    };
+  }
+
+  componentDidMount() {
+    InteractionManager.runAfterInteractions(this.fetchWeather());
```

Comment:
`...runAfterInteractions(this.fetchWeather)` or `...runAfterInteractions(() => this.fetchWeather())`;

The way it is now, it gets called immediately, not on callback in `runAfterInteractions`.